clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -228,16 +228,19 @@ class PointerVariableConstraint : public ConstraintVariable {
   // String representing declared bounds expression.
   std::string BoundsAnnotationStr;
 
+  // Does this variable represent a generic type?
+  // Generic types can be used with fewer restrictions.
+  bool IsGeneric;
```

Comment:
Would it make sense for this field to be in `ConstraintVariable` (the base class) rather than `PVConstraint`? I think the answer is probably no, since function pointers are also `PVConstraints`.

Reply:
As a field, no, I don't think that would make sense. It could be included as a virtual function, where PVConstraints implement it through this field, and function variables implement it by checking if any of their return/parameter ConstraintVariables are generic.

I guess that could come up if we ever wanted to handle sub typing with generic functions.

Reply:
Let's leave it.